src/vaults/Vault.ts

```diff
@@ -837,3 +928,154 @@ export class ExternalFileMultiContainer extends ExternalDirectoryContainer imple
+            indexData = JSON.parse(indexData);
+            const containerName = indexData.name;
+
+            if(!container || containerName === container){
+                const action = indexData.action.split('.');
+                const containerType = action[1];
+                const containerAction = action[2];
+
+                // Rebuild File containers
+                if(containerType.indexOf('_file') != -1) {
+
+                    // Multi-file containers
+                    if(containerType.indexOf('_multi') != -1){
+                        if(containerAction.indexOf('setsinglecontent') != -1) {
+
+                            if(!decryptedContainers[containerName]){
+                                decryptedContainers[containerName] = {};
+                            }
+
+                            if(subFile){
+                                if(subFile === indexData.params.fileName) {
+                                    decryptedContainers[containerName][subFile] = indexData.params.blob;
+                                    foundApplicableData = true;
+                                }
+                            }
+
+                            else {
+                                decryptedContainers[containerName][indexData.params.fileName] = indexData.params.blob;
+                                foundApplicableData = true;
+                            }
+                        }
+                    }
+
+                    // Single-file containers
+                    else {
+                        if (containerAction.indexOf('setcontents') != -1) {
+                            decryptedContainers[containerName] = indexData.params;
+                            foundApplicableData = true;
+                        }
+                    }
+
+                }
+
+                // Rebuild List containers
+                else if(containerType.indexOf('_list') != -1) {
+
+                    if(!decryptedContainers[containerName]){
+                        decryptedContainers[containerName] = [];
+                    }
+                    if(containerAction.indexOf('append') != -1) {
+                        decryptedContainers[containerName].push(indexData.params);
+                        foundApplicableData = true;
+                    }
+                }
+            }
+        }
+
+        if(!foundApplicableData){
+            throw new Error(`No data found for date`);
+        }
+
+        return decryptedContainers;
+    }
+
+    async decryptToDate(user: Wallet, container: string = null, date: string, subFile?: string) {
+        const toDate: Moment = moment(date,
+            [ExternalFileLedgerContainer.MOMENT_FORMAT, moment.ISO_8601],
+            true
+        ).add(1, "ms");
+
+        if(!toDate.isValid()){
+            throw new Error(`Invalid Date '${date}' not in format '${ExternalFileLedgerContainer.MOMENT_FORMAT}'`);
```

Comment:
Should we validate ISO-8601 format for `date` earlier than this?

Reply:
I had the date validation at the RPC layer originally, but that is ShipChain's entrypoint for this logic.  Other users may not have the same methods of interacting with the Vaults, so I opted for a validation specific to the Vault/Container.